socorro/unittest/schemas/test_schemas.py

```diff
@@ -25,21 +26,467 @@ def test_validate_schemas(reporoot):
+                "filename",
+            ],
+        ),
+        (
+            ".json_dump.crashing_thread.frames.[].registers.(re:^.+$)",
+            ["json_dump", "crashing_thread", "frames", "[]", "registers", "(re:^.+$)"],
+        ),
+    ],
+)
+def test_split_parts(path, expected):
+    assert list(split_path(path)) == expected
+
+
+class NonPublicParentError(Exception):
+    pass
+
+
+KNOWN_PUBLIC_FIELDS = {
+    "abort_message",
+    "accessibility",
+    "accessibility_client",
+    "accessibility_in_proc_client",
+    "adapter_device_id",
+    "adapter_driver_version",
+    "adapter_subsys_id",
+    "adapter_vendor_id",
+    "additional_minidumps",
+    "addons",
+    "addons_checked",
+    "address",
+    "android_board",
+    "android_brand",
+    "android_cpu_abi",
+    "android_cpu_abi2",
+    "android_device",
+    "android_display",
+    "android_fingerprint",
+    "android_hardware",
+    "android_manufacturer",
+    "android_model",
+    "android_version",
+    "app_init_dlls",
+    "app_notes",
+    "application_build_id",
+    "async_shutdown_timeout",
+    "available_page_file",
+    "available_physical_memory",
+    "available_virtual_memory",
+    "bios_manufacturer",
+    "build",
+    "client_crash_date",
+    "co_marshal_interface_failure",
+    "co_unmarshal_interface_result",
+    "completed_datetime",
+    "content_sandbox_capabilities",
+    "content_sandbox_capable",
+    "content_sandbox_enabled",
+    "content_sandbox_level",
+    "cpu_arch",
+    "cpu_info",
+    "cpu_microcode_version",
+    "crash_id",
+    "crash_report_keys",
+    "crash_time",
+    "crashing_thread",
+    "date_processed",
+    "distribution_id",
+    "dom_fission_enabled",
+    "dom_ipc_enabled",
+    "em_check_compatibility",
+    "flash_version",
+    "gmp_plugin",
+    "graphics_critical_error",
+    "graphics_startup_test",
+    "has_device_touch_screen",
+    "install_age",
+    "install_time",
+    "ipc_channel_error",
+    "ipc_fatal_error_msg",
+    "ipc_fatal_error_protocol",
+    "ipc_message_name",
+    "ipc_message_size",
+    "ipc_shutdown_state",
+    "ipc_system_error",
+    "is_garbage_collecting",
+    "java_exception",
+    "java_exception.exception",
+    "java_exception.exception.values",
+    "java_exception.exception.values.[]",
+    "java_exception.exception.values.[].stacktrace",
+    "java_exception.exception.values.[].stacktrace.frames",
+    "java_exception.exception.values.[].stacktrace.frames.[]",
+    "java_exception.exception.values.[].stacktrace.frames.[].filename",
+    "java_exception.exception.values.[].stacktrace.frames.[].function",
+    "java_exception.exception.values.[].stacktrace.frames.[].in_app",
+    "java_exception.exception.values.[].stacktrace.frames.[].lineno",
+    "java_exception.exception.values.[].stacktrace.frames.[].module",
+    "java_exception.exception.values.[].stacktrace.module",
+    "java_exception.exception.values.[].stacktrace.type",
+    "java_exception_raw",
```

Comment:
This is marked protected, so it (and it's children) shouldn't be here. It's a ref, so I think the bug is around traversing refs.